test/forkMainnet/SmartOrderStrategy.t.sol

```diff
@@ -0,0 +1,255 @@
+
+        sosInputToken.assertChange(-int256(defaultInputAmount));
+        gsOutputToken.assertChange(int256(expectedOut));
+    }
+
+    function testUniswapV2WithAmountReplace() public {
+        bytes memory uniswapData = abi.encodeWithSelector(
+            IUniswapRouterV2.swapExactTokensForTokens.selector,
+            defaultInputAmount,
+            0,
+            defaultUniV2Path,
+            address(smartOrderStrategy),
+            defaultExpiry
+        );
+        ISmartOrderStrategy.Operation[] memory operations = new ISmartOrderStrategy.Operation[](1);
+        operations[0] = ISmartOrderStrategy.Operation({
+            dest: UNISWAP_V2_ADDRESS,
+            inputToken: defaultInputToken,
+            inputRatio: defaultInputRatio,
+            dataOffset: uint128(4 + 32), // add 32 bytes of length prefix
+            value: 0,
+            data: uniswapData
+        });
+        bytes memory data = abi.encode(operations);
+
+        // get the exact quote from uniswap
+        uint256 inputAmountAfterRatio = (defaultInputAmount * defaultInputRatio) / Constant.BPS_MAX;
+        IUniswapRouterV2 router = IUniswapRouterV2(UNISWAP_V2_ADDRESS);
+        uint256[] memory amounts = router.getAmountsOut(inputAmountAfterRatio, defaultUniV2Path);
+        uint256 expectedOut = amounts[amounts.length - 1];
+
+        vm.startPrank(genericSwap, genericSwap);
+        IERC20(defaultInputToken).safeTransfer(address(smartOrderStrategy), defaultInputAmount);
+        Snapshot memory sosInputToken = BalanceSnapshot.take(address(smartOrderStrategy), defaultInputToken);
+        Snapshot memory gsOutputToken = BalanceSnapshot.take(genericSwap, defaultOutputToken);
+        smartOrderStrategy.executeStrategy(defaultInputToken, defaultOutputToken, defaultInputAmount, data);
+        vm.stopPrank();
+
+        sosInputToken.assertChange(-int256(inputAmountAfterRatio));
+        gsOutputToken.assertChange(int256(expectedOut));
+    }
+
+    function testUniswapV2WithMaxAmountReplace() public {
+        bytes memory uniswapData = abi.encodeWithSelector(
+            IUniswapRouterV2.swapExactTokensForTokens.selector,
+            defaultInputAmount,
+            0,
+            defaultUniV2Path,
+            address(smartOrderStrategy),
+            defaultExpiry
+        );
+        ISmartOrderStrategy.Operation[] memory operations = new ISmartOrderStrategy.Operation[](1);
+        operations[0] = ISmartOrderStrategy.Operation({
+            dest: UNISWAP_V2_ADDRESS,
+            inputToken: defaultInputToken,
+            inputRatio: Constant.BPS_MAX, // BPS_MAX indicate the input amount will be replaced by the actual balance
+            dataOffset: uint128(4 + 32), // add 32 bytes of length prefix
+            value: 0,
+            data: uniswapData
+        });
+        bytes memory data = abi.encode(operations);
+
+        // get the exact quote from uniswap
+        uint256 actualInputAmount = defaultInputAmount + 1234;
```

Comment:
What's the implication of adding `1234` here?

Reply:
It's just a random new input amount. Changed to `5678` without `defaultInputAmount` in 6a93eab.